src/tildePrefix.ml

```diff
@@ -37,46 +37,123 @@
+let extract_tilde_prefix_from_word_if_present (word : word_cst) : word_cst =
+  match merge_leading_literals word with
+  | WordLiteral literal :: word when starts_with_tilde literal ->
+    extract_tilde_prefix_from_literal literal @ word
+  | word -> word
+
+(** Extracts the first and last elements of a list; returns a triple consisting
+    of the first element, the “body” of the list and the last element. Assumes
+    that the list has at least two elements or raises [Invalid_arg]. *)
+let extract_list_head_and_foot (list : 'a list) : 'a * 'a list * 'a =
+  let rec extract_list_foot (list : 'a list) : 'a list * 'a =
+    match list with
+    | [] -> assert false
+    | [foot] -> ([], foot)
+    | first :: rest ->
+      let (body, foot) = extract_list_foot rest in
+      (first :: body, foot)
+  in
+  match list with
+  | [] | [_] -> invalid_arg "extract_list_head_and_foot"
+  | head :: rest ->
+    let (body, foot) = extract_list_foot rest in
+    (head, body, foot)
+
+(** Splits the given word on each literal colon character, returning a list of
+    words. A word semantically equivalent can be re-obtained by interspersing a
+    [WordLiteral ":"] between all the words. *)
+(* REVIEW: this might make sense on its own as one of the CST helpers. *)
+let split_word_on_colon (word_to_process : word_cst) : word_cst list =
+  let rec split_word_component_on_colon
+      ~(processed_words_rev : word_cst list)
+      ~(current_word_rev : word_cst)
+      ~(word_to_process : word_cst)
+    : word_cst list
+    =
+    match word_to_process with
+    | WordLiteral literal :: word_to_process when String.contains literal ':' ->
+      let subliterals = String.split_on_char ':' literal in
+      let (first, body, foot) = extract_list_head_and_foot subliterals in
+      let processed_words_rev =
+        List.rev_map (fun literal -> [WordLiteral literal]) body
+        @ [List.rev (WordLiteral first :: current_word_rev)]
+        @ processed_words_rev
+      in
+      split_word_component_on_colon
+        ~processed_words_rev
+        ~current_word_rev:[WordLiteral foot]
+        ~word_to_process
+
+    | word_component :: word_to_process ->
+      split_word_component_on_colon
+        ~processed_words_rev
+        ~current_word_rev:(word_component :: current_word_rev)
+        ~word_to_process
+
+    | [] ->
+      List.rev ((List.rev current_word_rev) :: processed_words_rev)
+  in
+  split_word_component_on_colon
+    ~processed_words_rev:[]
+    ~current_word_rev:[]
+    ~word_to_process
+
+(** Concatenates the given list of words into one, interspersing it with literal
+    colon characters. *)
+let rec concat_words_with_colon (words : word_cst list) : word_cst =
+  match words with
+  | [] -> []
+  | [word] -> word
+  | word :: words -> word @ [WordLiteral ":"] @ concat_words_with_colon words
+
+(** Recognises tilde prefixes in a word. The [rhs_assignment] parameter
+    influences the behaviour of the recognition as described by the standard. *)
```

Comment:
Let's try to refer to the standard section title or to quote the paragraph of the standard to ease code review!